packages/actor-rdf-parse-html/lib/ActorRdfParseHtml.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  ActorRdfParse,
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, IActorTest, Mediator} from "@comunica/core";
+import {Readable} from "stream";
+
+/**
+ * A HTML RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract JSON-LD from HTML files and parse the JSON-LD based RDF serializations
+ * and announce the presence of them by media type.
+ */
+// Gebaseerd op JSON LD parser
+export class ActorRdfParseHtml extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+      Promise<IActorRdfParseOutput> {
+
+    const htmlString: string = await require('stream-to-string')(action.input);
+
+    // JSON-LD extraction
+    let jsonString: string = '';
+    const DOMParser = require('xmldom').DOMParser;
+    const doc = new DOMParser().parseFromString(htmlString, 'text/html');
+    const scripts = doc.getElementsByTagName('script');
+    for (let i = 0; i < scripts.length; i++) {
+      if (scripts[i].getAttribute("type") === "application/ld+json") {
```

Comment:
This is hardcoded to only support JSON-LD, while any other (supported) RDF serialization should be parseable as well (like Turtle, TriG, ...)

Reply:
The assigned task here was to only parse the JSON-LD from between the script tags.

Reply:
I thought this was the PR for #138.

@pietercolpaert Did I misunderstand this?

Reply:
I first limited @astverha’s task to JSON-LD snippets, to then gradually expand to others as well! First, let’s see how this PR looks for JSON-LD. There should indeed be other content-types there!

Reply:
Ok, then I'll wait for merging this until it's feature-complete :-)